Configure

```diff
@@ -2522,6 +2539,22 @@ sub which
     }
 }
 
+sub env
+{
+    my $name = shift;
+    my $fn = sub {
+        # Note that if $ENV{$name} doesn't exist or is undefined,
+        # $config{perlenv}->{$name} will be created with the value
+        # undef.  This is intentional.
+
+        $config{perlenv}->{$name} = $ENV{$name}
+            if ! exists $config{perlenv}->{$name};
+        return $config{perlenv}->{$name};
+    };
+
+    return $reading_targets ? $fn : $fn->();
```

Comment:
The problem I have with this approach is that it's another detail/dependency you have to keep in mind/maintain. In other words I'd say that *less* "strange" dependencies is better. [Which by the way the idea behind "target factory" I've discussed in private in #4684 context, it's an attempt to unify member instantiation without member-specific hooks.] Also note that most invocations of env are already lazy. Because most of them appear in lazily-called subs. So wouldn't it be more appropriate to say that if any of remaining calls, i.e. those not currently lazy, has to be lazy for some reason, then it could be made lazy [by placing it into anonymous sub]?

Reply:
Ahhh, good point.  Okie, will redo this.